src/pybind/mgr/pg_autoscaler/module.py

```diff
@@ -210,6 +214,60 @@ def shutdown(self):
         self.log.info('Stopping pg_autoscaler')
         self._shutdown.set()
 
+    def check_pool_activity(check_time, pool_stats, pool_id, final_pg_target, old_values):
+        total = (int(pool_stats.get("rd", 0)) + int(pool_stats.get("wr", 0)))
+        if pool_id in old_values.keys():
+            if check_time == 1:
+                if old_values[pool_id]['times'] == total:
+                    old_values[pool_id]['unused'] += 1
+                    old_values[pool_id]['add'] = 0
+                    old_values[pool_id]['actual_add'] = 0
+                    if final_pg_target - old_values[pool_id]['unused'] < PG_NUM_MIN:
+                        old_values[pool_id]['unused'] -= 1
+                    else:
+                        for k1 in old_values:
+                            if old_values[k1]['unused'] == 0 and pool_id != k1:
+                                old_values[k1]['add'] += 1
+                else:
+                    if old_values[pool_id]['unused'] != 0:
+                        old_values[pool_id]['unused'] = 0
+                        old_values[pool_id]['actual_unused'] = 0
+                    else:
+                        for k2 in old_values:
+                            while old_values[k2]['add'] != 0 and old_values[pool_id]['unused'] != 0:
+                                if k2 != old_values[pool_id]:
```

Comment:
Should the conditional be placed in the inner loop?  ie won't we keep spinning if k2 == old_values[pool_id] and the while conditions are not met? Also, can we avoid the inner while loop entirely by looking at the min of add and unused? ie something like:

```
for k2 in old_values:
    # short circuit the outer for loop when we are done
    if old_values[pool_id]['unused'] == 0:
        break
    # continue when there's nothing to do for k2
    if k2 == old_values[pool_id] || old_values[k2]['add']  == 0:
        continue 
    change = min(old_values[k2]['add'], old_values[pool_id]['unused'])
    old_values[k2]['add'] -= change
    old_values[pool_id]['unused'] -= change
```

Reply:
look more into this